lib/runner.go

```diff
@@ -194,79 +219,86 @@ func (r *Runner) runAllTests(testFiles []string, testFolder string) {
 }
 
 func (r *Runner) runSingleTest(testFile string, testFolder string) TestResult {
-	command := exec.Command(filepath.Join(testFolder, testFile))
+	ctx, cancel := context.WithTimeout(context.Background(), r.timeout)
+	defer cancel()
+
+	testPath := filepath.Join(testFolder, testFile)
+
+	command := exec.CommandContext(ctx, testPath)
+
 	commandOutput := &bytes.Buffer{}
 	command.Stdout = commandOutput
 	command.Stderr = commandOutput
 
-	// Propagate timeout information from brain to script, via the
-	// environment of the script.
+	// Propagate timeout information from brain to script, via the environment of the script.
 	env := os.Environ()
-	env = append(env, fmt.Sprintf("TESTBRAIN_TIMEOUT=%d", int(r.Timeout.Seconds())))
+	env = append(env, fmt.Sprintf("TESTBRAIN_TIMEOUT=%d", int(r.timeout.Seconds())))
```

Comment:
why not just `env = append(env, fmt.Sprintf("TESTBRAIN_TIMEOUT=%f", r.timeout.Seconds()))`?

Reply:
Thanks. This line I was merely renaming the previously exported field. I'll do this.